components/apply/AppForm.tsx

```diff
@@ -178,7 +178,7 @@ const AppForm: FC<Props> = ({
           secondChoiceRole: values.secondChoiceRole || "",
           shortAnswerQuestions: values.shortAnswerQuestions,
           roleSpecificQuestions: values.roleSpecificQuestions,
-          timestamp: Date.now(),
+          timestamp: serverTimestamp(),
```

Comment:
We want to use the server's time rather than the client-side time so that we can be sure the timestamp is accuracte.